tsl/src/nodes/skip_scan/planner.c

```diff
@@ -367,14 +408,30 @@ skip_scan_path_create(PlannerInfo *root, IndexPath *index_path, double ndistinct
 	Var *var = get_distinct_var(root, index_path, skip_scan_path);
 
 	if (!var)
+	{
 		return NULL;
+	}
 
 	skip_scan_path->distinct_var = var;
 
 	/* build skip qual this may fail if we cannot look up the operator */
 	if (!build_skip_qual(root, skip_scan_path, index_path, var))
 		return NULL;
 
+	if (skip_scan_path->is_compressed)
+	{
+		List *physical_tlist = NULL;
+		physical_tlist = build_physical_tlist(root, index_path->path.parent);
+		if (physical_tlist)
+		{
+			PathTarget *ptarget = create_pathtarget(root, physical_tlist);
+			skip_scan_path->cpath.path.pathtarget = ptarget;
+			index_path->path.pathtarget = ptarget;
+		}
```

Comment:
Let's add a comment explaining why we have to discard the planned targetlist and replace it with the physical.